src/components/TaskclusterAuth/TaskclusterCallback.tsx

```diff
@@ -0,0 +1,129 @@
+import { useEffect, useState } from 'react';
+
+import { useSearchParams } from 'react-router-dom';
+
+// TODO remove this after integrating the auth code logic
+localStorage.setItem('requestState', 'OkCrH5isZncYqeJbRDelN');
+localStorage.setItem('tcRootUrl', 'https://firefox-ci-tc.services.mozilla.com');
+// end TODO remove
+
+const tcAuthCallbackUrl = '/taskcluster-auth';
+const redirectURI = `${window.location.origin}${tcAuthCallbackUrl}`;
+// TODO update to a valid perfcompare clientId
+const clientId = `treeherder-localhost-5000-client`;
+
+interface RequestOptions {
+  method: string;
+  body: string;
+  headers: {
+    'Content-Type': string;
+  };
+}
+
+// from treeherder
+const processErrorMessage = async (error: string, status) => {
+  let errorMessage = '';
+
+  if (status >= 500) {
+    errorMessage +=
+      'There was a problem retrieving the data. Please try again in a minute.';
+  }
+
+  if (status === 400) {
+    errorMessage += 'The action resulted in a bad request.';
+  }
+
+  if (error instanceof Object) {
+    const key = Object.keys(error);
+
+    errorMessage += ` ${key}: ${error[key]}`;
+  } else if (error) {
+    errorMessage += error;
+  }
+  return errorMessage ? errorMessage.trim() : error;
+};
+
+// from treeherder, TODO move this in another file/folder
```

Comment:
Maybe move it to src/logic/treeherder.ts?

Reply:
I'd move all the "gory" things related to requests to something such as logic/taskcluster.ts